c++/src/kj/memory.h

```diff
@@ -22,12 +22,31 @@
 #pragma once
 
 #include "common.h"
-#ifdef KJ_DEBUG
-#include <atomic>  // std::atomic for KJ_ASSERT_PTR_COUNTERS
-#endif
 
 KJ_BEGIN_HEADER
 
+// KJ_DEBUG_MEMORY == 1 enables variety of checks designed to catch memory usage errors.
+// KJ_DEBUG_MEMORY == 0 disables all such checks.
+// When not set it is initialized to 1 in KJ_DEBUG mode, 0 otherwise.
+#if !defined(KJ_DEBUG_MEMORY)
+#if defined(KJ_DEBUG)
+#define KJ_DEBUG_MEMORY 1
+#else
+#define KJ_DEBUG_MEMORY 0
+#endif // KJ_DEBUG
+#endif // KJ_DEBUG_MEMORY
+
+// KJ_ASSERT_PTR_COUNTERS == 1 keeps track of active Ptr<T> instances and asserts validity
+// of their ownership.
+// Enabled by KJ_DEBUG_MEMORY by default.
+#if !defined(KJ_ASSERT_PTR_COUNTERS)
+#define KJ_ASSERT_PTR_COUNTERS KJ_DEBUG_MEMORY
+#endif
+
+#if KJ_ASSERT_PTR_COUNTERS
+#include <atomic>  // std::atomic for KJ_ASSERT_PTR_COUNTERS
```

Comment:
@kentonv I am not sure what role does KJ_BEGIN_HEADER play? Can I have `#include <atomic>`  here?

Reply:
Is it ok to put these defines before KJ_BEGIN_HEADER then?

Reply:
(You can place your `#if` machinery above `KJ_BEGIN_HEADER` as well, just try to write it so it won't generate any warnings even with pedantic settingcs.)

Reply:
done. 🤞